docs/guides/src/main/server/vault.adoc

```diff
@@ -0,0 +1,59 @@
+<#import "/templates/guide.adoc" as tmpl>
+<#import "/templates/kc.adoc" as kc>
+
+<@tmpl.guide
+title="Using Kubernetes Secrets"
+summary="Learn how to use Kubernetes / OpenShift secrets in Keycloak"
+priority=30
+includedOptions="vault vault-*">
+
+Keycloak supports a file based vault implementation for Kubernetes / OpenShift secrets. Mount Kubernetes secrets as data volumes, and they appear in a directory with a flat-file structure. Keycloak represents each secret as a file with the file’s name as the secret name and the file’s content as the secret value.
+
+== Available integrations
+You can use Kubernetes / OpenShift secrets for the following use-cases:
+
+* Obtain the SMTP Mail server Password
+* Obtain the LDAP Bind Credential when using LDAP-based User Federation
+* Obtain the OIDC identity providers Client Secret when integrating external identity providers
+
+== Enabling the vault
+Enable the file based vault by building Keycloak using the following build option:
+
+<@kc.build parameters="--vault=file"/>
+
+== Setting the base directory to lookup secrets
+Kubernetes / OpenShift secrets are basically mounted files, so you have to configure a directory for these files to be mounted in:
+
+<@kc.start parameters="--vault-dir=/my/path"/>
+
+== Realm-specific secret files
+Kubernetes / OpenShift Secrets are used per-realm basis in Keycloak, so there's a naming convention for the file in place:
```

Comment:
I think we need to explain a little better how to name the `data` fields into the Secrets, since it can be quite challenging to align all the strings to match.
Here you have an example with a working configuration:
https://issues.redhat.com/browse/SSOSUP-105?focusedCommentId=19404192&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-19404192